### ISSUE TITLE
Undeclared dependency in `AbstractTraceableHttpClient`

### ISSUE THREAD

**Issue by Dormilich** [getsentry/sentry-symfony#714]
### Environment

PHP: 7.4.33
sentry/sdk: (omitted)
sentry/sentry: 3.17.0
sentry/sentry-symfony: 4.7.0
symfony/*: 5.4.21

### Steps to Reproduce

Note: Installation done through composer

1. install Symfony LTS (5.4)
2. install _symfony/http-client_ and _nyholm/psr7_
3. disable _sentry/sdk_ in `composer.json`
4. install _sentry/sentry-symfony_
5. use default configuration (tracing enabled)
6. trigger the error by running code that makes an HTTP request

### Expected Result

The HTTP request should be processed without error.

### Actual Result

The requests throws a class-not-found error.

```
NOTICE: PHP message: php.CRITICAL: Uncaught Error: Class 'GuzzleHttp\Psr7\Uri' not found {"exception":"[object] (Error(code: 0): Class 'GuzzleHttp\\Psr7\\Uri' not found at /app/vendor/sentry/sentry-symfony/src/Tracing/HttpClient/AbstractTraceableHttpClient.php:54)"} []
```

The reason for this error is that `GuzzleHttp\Psr7\Uri` is used in `AbstractTraceableHttpClient::request()` without declaring it a dependency in either _sentry/sentry-symfony_ or _sentry/sentry_. However, _guzzlehttp/psr7_ is a dependency of _sentry/sdk_.

A possible fix were to have a dependency on `Psr\Http\Message\UriFactoryInterface` in `AbstractTraceableHttpClient` since that is already required by _sentry/sentry_.

**Comment by cleptric** [comment]
I'll prepare a release that includes the fix for this issue (https://github.com/getsentry/sentry-symfony/pull/708). Sorry about that!

**Comment by Dormilich** [comment]
Meh, didn't think of looking up closed issues, sorry.

**Comment by Dormilich** [comment]
I would prefer if this were solved without a direct dependency on _guzzlehttp/psr7_ as the affected code does not require a specific implementation to achieve its goal (`$partialUri` could be created from any `UriInterface` instance).

**Comment by Jean85** [comment]
@cleptric we should avoid similar issues in the future by adding https://packagist.org/packages/maglnet/composer-require-checker in CI

**Comment by cleptric** [comment]
Released via [4.8.0](https://github.com/getsentry/sentry-symfony/releases/tag/4.8.0)
